From patchwork Fri Jan 25 11:19:33 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 10781209
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A974E1850
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:19:58 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 97D2B2F59F
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:19:58 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 8BA9D2F5B3; Fri, 25 Jan 2019 11:19:58 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 31DA72F59F
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:19:58 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727895AbfAYLTs (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 25 Jan 2019 06:19:48 -0500
Received: from mail-wm1-f68.google.com ([209.85.128.68]:37212 "EHLO
        mail-wm1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727745AbfAYLTq (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 25 Jan 2019 06:19:46 -0500
Received: by mail-wm1-f68.google.com with SMTP id g67so6265422wmd.2
        for <kvm@vger.kernel.org>; Fri, 25 Jan 2019 03:19:45 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20130820;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=OymLaeZkDqCUBdn/5EkTjADbVQG+txQBjbSU5EvNNfY=;
        b=ajqikQFf6pHhrHJ3KnMmGkr46uu8WWGjh0ZENtlIXi+NMHoxdmLtExtN+kVeNBPPxL
         0LnVt7o49xHWd3fWTAsCA/05Emo3/r3TTcxNmkjUOqNYm3V9cvBH4fJBLY5BeETCZ/62
         x+9rymGRPWPfUEMOZ7KDD6rLlEpQ7GVEpqfwKZbfQdCMoky9pQVmnHasc+Wn3NHEZlk0
         QzxB32oTotLwrJCUOxZ/37VoW32KlAgyWMErsMgMxhGgwjGUW0GbN8xQ9a3aW02cGCt7
         GBKUIwNhyv+RhD9fH04MujWuRi6XWQhBzqJNK4ecCNdr4HC/5eg+yf6lgHGWRheyoYix
         ZpDA==
X-Gm-Message-State: AJcUukc8IPPLFsjveNpzKqaD2dAvSWp14QVTCqHQCnKWW6nHGSagK9cA
        8vN6Gfu0u6ptMwOOyaZ2/FJBBvyWxghcxg==
X-Google-Smtp-Source: 
 ALg8bN5uE6ngqEwW+Qx72iqaWsKkJEUmcEBRBUD+aZ1x1+As4rwbP4PB2NQIF+uNhtkfUbcqbS7NeA==
X-Received: by 2002:a1c:e383:: with SMTP id a125mr944968wmh.1.1548415183845;
        Fri, 25 Jan 2019 03:19:43 -0800 (PST)
Received: from vitty.brq.redhat.com.cz ([147.229.117.36])
        by smtp.gmail.com with ESMTPSA id
 f2sm98101713wru.14.2019.01.25.03.19.42
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Fri, 25 Jan 2019 03:19:43 -0800 (PST)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 Roman Kagan <rkagan@virtuozzo.com>, Liran Alon <liran.alon@oracle.com>,
 linux-kernel@vger.kernel.org
Subject: [PATCH v2 1/2] x86/kvm/hyper-v: don't recommend doing reset via
 synthetic MSR
Date: Fri, 25 Jan 2019 12:19:33 +0100
Message-Id: <20190125111934.28782-2-vkuznets@redhat.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190125111934.28782-1-vkuznets@redhat.com>
References: <20190125111934.28782-1-vkuznets@redhat.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

System reset through synthetic MSR is not recommended neither by genuine
Hyper-V nor my QEMU.

Fixes: 2bc39970e932 ("x86/kvm/hyper-v: Introduce KVM_GET_SUPPORTED_HV_CPUID")
Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Reviewed-by: Liran Alon <liran.alon@oracle.com>
---
 arch/x86/kvm/hyperv.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/x86/kvm/hyperv.c b/arch/x86/kvm/hyperv.c
index ac44a681f065..4840f5b3c88f 100644
--- a/arch/x86/kvm/hyperv.c
+++ b/arch/x86/kvm/hyperv.c
@@ -1847,7 +1847,6 @@ int kvm_vcpu_ioctl_get_hv_cpuid(struct kvm_vcpu *vcpu, struct kvm_cpuid2 *cpuid,
 		case HYPERV_CPUID_ENLIGHTMENT_INFO:
 			ent->eax |= HV_X64_REMOTE_TLB_FLUSH_RECOMMENDED;
 			ent->eax |= HV_X64_APIC_ACCESS_RECOMMENDED;
-			ent->eax |= HV_X64_SYSTEM_RESET_RECOMMENDED;
 			ent->eax |= HV_X64_RELAXED_TIMING_RECOMMENDED;
 			ent->eax |= HV_X64_CLUSTER_IPI_RECOMMENDED;
 			ent->eax |= HV_X64_EX_PROCESSOR_MASKS_RECOMMENDED;

From patchwork Fri Jan 25 11:19:34 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 10781211
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 9DC5B1399
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:20:00 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 8CA0F2F59F
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:20:00 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 7FF802F5B3; Fri, 25 Jan 2019 11:20:00 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 2D80F2F59F
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 11:20:00 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727826AbfAYLTs (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 25 Jan 2019 06:19:48 -0500
Received: from mail-wr1-f68.google.com ([209.85.221.68]:43207 "EHLO
        mail-wr1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726311AbfAYLTr (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 25 Jan 2019 06:19:47 -0500
Received: by mail-wr1-f68.google.com with SMTP id r10so9849637wrs.10
        for <kvm@vger.kernel.org>; Fri, 25 Jan 2019 03:19:46 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=g6yhpZQzcn+9JBMSIZS6widjg9wc9aPVmdpk6Zov9nE=;
        b=Lb6Jn5WQxrSaPzWhEbmL+kuVlDJZb/w3+6j4pC6oo2xG7RGm1sVR2ruT3LeJxTQcXZ
         YWqv/QifmpxSC27ui6Nxyz4+tLOWtf6eySgS5rCE4LieNqFi5QROa5I5SU4KbvfM8VLn
         KLey1NOOLm6cCPs3HEZ/TCbVl0nPQpazouzP+0BDLG9r5UhyRtac2R14cOf2R+JgApP+
         6Nf5smYAnT6QDzxUdCqawpCycvulm0AlTtoV7SJAb1cfOcdxjFYLsP/2uWrN+LpbngHd
         zBy0zqhAL1rYfAYUczcd718j7RpRY+m/nvrWUotuQVj/ZUVo94tj64DxcTeWhHl6hbpk
         017Q==
X-Gm-Message-State: AJcUukeXSi/6UeXjGYt7dlFQjGvL0ZNyGQbwEsm+APBK1OAx1Z90r7nj
        /UdzcA45MAwKj8sAAPb/64rxCQIGjSyGDg==
X-Google-Smtp-Source: 
 ALg8bN6/iIA6z/ESY3qBrqYi6U8TPXZ9BzrqgmoL2bA2GXLIw9p7+nC9mQ5m4FZLq1/NytiyPqxA5w==
X-Received: by 2002:a5d:6549:: with SMTP id z9mr10750246wrv.116.1548415184997;
        Fri, 25 Jan 2019 03:19:44 -0800 (PST)
Received: from vitty.brq.redhat.com.cz ([147.229.117.36])
        by smtp.gmail.com with ESMTPSA id
 f2sm98101713wru.14.2019.01.25.03.19.43
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Fri, 25 Jan 2019 03:19:44 -0800 (PST)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 Roman Kagan <rkagan@virtuozzo.com>, Liran Alon <liran.alon@oracle.com>,
 linux-kernel@vger.kernel.org
Subject: [PATCH v2 2/2] x86/kvm/hyper-v: recommend using eVMCS only when it is
 enabled
Date: Fri, 25 Jan 2019 12:19:34 +0100
Message-Id: <20190125111934.28782-3-vkuznets@redhat.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190125111934.28782-1-vkuznets@redhat.com>
References: <20190125111934.28782-1-vkuznets@redhat.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

We shouldn't probably be suggesting using Enlightened VMCS when it's not
enabled (not supported from guest's point of view). Hyper-V on KVM seems
to be fine either way but let's be consistent.

Reviewed-by: Liran Alon <liran.alon@oracle.com>
Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
---
 arch/x86/kvm/hyperv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kvm/hyperv.c b/arch/x86/kvm/hyperv.c
index 4840f5b3c88f..4730fcaa70cf 100644
--- a/arch/x86/kvm/hyperv.c
+++ b/arch/x86/kvm/hyperv.c
@@ -1850,7 +1850,8 @@ int kvm_vcpu_ioctl_get_hv_cpuid(struct kvm_vcpu *vcpu, struct kvm_cpuid2 *cpuid,
 			ent->eax |= HV_X64_RELAXED_TIMING_RECOMMENDED;
 			ent->eax |= HV_X64_CLUSTER_IPI_RECOMMENDED;
 			ent->eax |= HV_X64_EX_PROCESSOR_MASKS_RECOMMENDED;
-			ent->eax |= HV_X64_ENLIGHTENED_VMCS_RECOMMENDED;
+			if (evmcs_ver)
+				ent->eax |= HV_X64_ENLIGHTENED_VMCS_RECOMMENDED;
 
 			/*
 			 * Default number of spinlock retry attempts, matches
